thunderbird/wps_io.py

```diff
@@ -0,0 +1,53 @@
+from pywps import LiteralInput, ComplexInput, ComplexOutput, FORMATS
+
+import logging
+
+
+def ncInput(identifier, title, abstract, max_occurs):
+    return ComplexInput(
+        identifier,
+        title,
+        abstract=abstract,
+        min_occurs=1,
+        max_occurs=max_occurs,
+        supported_formats=[FORMATS.NETCDF, FORMATS.DODS],
+    )
```

Comment:
I appreciate the factoring out you did here, but I think this may be a case where it was not necessary.  We are basically defining the exact same params we were before just through a method this time.  Looking at this, two options come to mind:
1. We scrap this method and just define the complex netcdf inputs the way we were before.  Each of the inputs are unique enough that we cannot generalize the creation process anymore than using the object constructor.

2. Parameterize each of the attributes with default values.  This gives us a default object if we want, but also gives us the flexibility to customize some aspects.  Something like so:

```python3
def ncInput(identifier='netcdf', title='NetCDF', abstract='netcdf file', min_occurs=1, max_occurs=MAX_OCCURS, supported_formats=[FORMATS.NETCDF, FORMATS.DODS]):
    return ComplexInput(
        identifier=identifier,
        title=title,
        abstract=abstract,
        min_occurs=min_occurs,
        max_occurs=max_occurs,
        supported_formats=supported_formats,
    )
```
I'm leaning towards option 1, but I will leave it up to your discretion.

Reply:
Yeah, I think option 1 is better. Pretty much the only thing that's constant across the inputs is `min_occurs` and `supported_formats`.